spec/models/review_spec.rb

```diff
@@ -44,4 +45,101 @@
   it "can create a review" do
     expect(Fabricate(:review)).to be_truthy
   end
+
+  it "can not create duplicate facility reviews" do
+    review = Fabricate(:review)
+    facility = Fabricate(:facility)
+    Fabricate(
+      :facility_review,
+      review: review,
+      facility: facility,
+      experience: :was_not_allowed
+    )
+    expect(review.facility_reviews.count).to eq(1)
+
+    # Generate a facility review for the same facility, should raise an error:
+    expect do
+      Fabricate(
+        :facility_review,
+        review: review,
+        facility: facility,
+        experience: :was_not_allowed
+      )
+    end.to raise_error(ActiveRecord::RecordInvalid)
+
+    expect(review.facility_reviews.count).to eq(1)
+  end
+end
+
+RSpec.describe Review, type: :request do
+  let(:user) { Fabricate(:user) }
+  let(:review) { Fabricate(:review, user: user) }
+  let(:space) { review.space }
+  let(:facility) { Fabricate(:facility) }
+  let(:facility_review) do
+    Fabricate(
+      :facility_review,
+      space: space,
+      user: user,
+      review: review,
+      facility: facility,
+      experience: :was_not_allowed
+    )
+  end
+
+  before do
+    sign_in user
+  end
+
+  it "can load the edit path" do
+    get edit_review_path(review)
+    expect(response).to have_http_status(:success)
+  end
+
+  it "has loaded the facility review, and set it to unlikely" do
+    facility_review.reload
+    expect(review.facility_reviews.count).to eq(1)
+    space.aggregate_facility_reviews
+    expect(space.reviews_for_facility(facility)).to eq("unlikely")
+  end
+
+  it "will update the facility review if the Review is updated" do
+    patch review_path(review), params: {
```

Comment:
Å bruke "patch" her, i stedet for å fylle ut formen og trykke submit, tror jeg er riktig - men hadde vært digg å ha en test av å trykke ut formen også. 

Problemet med å patch er at man må litt painstakenly konstruere en PATCH request lik som den form-en sender ut, som på linje 108 og utover her.

Reply:
Hvis vi vil ha en test som trykker på formen går vi over til feature tests som er trege og tar mye tid i vedlikehold på hver endring

Reply:
Skjønner! Takk